vmngclient/tests/test_logs_api.py

```diff
@@ -0,0 +1,35 @@
+import unittest
+from pathlib import Path
+from unittest.mock import patch
+
+from vmngclient.api.logs_api import LogsAPI
+
+
+class MyTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.logs = [
+            {"logid": "12345", "logmessage": "message from log 1", "tenant": "tenant234", "entry_time": 1671028426000},
+            {"logid": "67890", "logmessage": "message from log 2", "tenant": "tenant987", "entry_time": 1671028426000},
+        ]
+
+    @patch("vmngclient.session.vManageSession")
+    def test_get_auditlogs_file_path_not_provided(self, mock_session):
+        # Arrange
+        mock_session.get_data.return_value = self.logs
+        default_file_path = Path(__file__).parents[1].joinpath("api").joinpath("audit.log")
+        # Act
+        LogsAPI(mock_session).get_auditlogs()
+        does_file_exist = default_file_path.is_file()
+        # Assert
+        self.assertTrue(does_file_exist)
+
+    @patch("vmngclient.session.vManageSession")
+    def test_get_auditlogs_file_path_provided(self, mock_session):
+        # Arrange
+        mock_session.get_data.return_value = self.logs
+        file_path = Path(__file__).parents[0].joinpath("test_file.log")
+        # Act
+        LogsAPI(mock_session).get_auditlogs(file_path=file_path)
+        does_file_exist = file_path.is_file()
+        # Assert
+        self.assertTrue(does_file_exist)
```

Comment:
can be parameterized

Reply:
it can, but will look like this:
```python
    @parameterized.expand([[None], ["test_file.log"]])
    @patch("vmngclient.session.vManageSession")
    def test_get_auditlogs_file_path(self, file_name, mock_session):
        # Arrange
        mock_session.get_data.return_value = self.logs
        if file_name:
            default_file_path = Path(__file__).parents[0].joinpath(file_name)
            file_path = str(Path(__file__).parents[0].joinpath(file_name))
        else:
            default_file_path = Path(__file__).parents[1].joinpath("api").joinpath("audit.log")
            file_path = None
        # Act
        if file_name:
            LogsAPI(mock_session).get_auditlogs(file_path=file_path)
        else:
            LogsAPI(mock_session).get_auditlogs()
        does_file_exist = default_file_path.is_file()
        # Assert
        self.assertTrue(does_file_exist)
```
is it really better?

Reply:
I prefer not parametrized :)